File: tell-dont-ask-kata-kotlin/src/test/java/it/gabrieletondi/telldontaskkata/doubles/TestOrderRepository.kt
package test.java.it.gabrieletondi.telldontaskkata.doubles



import main.java.it.gabrieletondi.telldontaskkata.domain.Order
import main.java.it.gabrieletondi.telldontaskkata.repository.OrderRepository
import java.util.ArrayList

class TestOrderRepository : OrderRepository {
    var savedOrder: Order? = null
        private set
    private val orders = ArrayList<Order>()

    override fun save(order: Order) {
        this.savedOrder = order
    }

    override fun getById(orderId: Int): Order {
        return orders.stream().filter { o -> o.id == orderId }.findFirst().get()
    }

    fun addOrder(order: Order) {
        this.orders.add(order)
    }
}

File: tell-dont-ask-kata-csharp/TellDontAsk.Tests/doubles/TestOrderRepository.cs
using System.Collections.Generic;
using System.Linq;

public class TestOrderRepository : OrderRepository
{
    private Order insertedOrder;
    private List<Order> orders = new List<Order>();

    public Order GetSavedOrder()
    {
        return insertedOrder;
    }

    public void Save(Order order)
    {
        insertedOrder = order;
    }

    public Order GetById(int orderId)
    {
        return orders.First(o => o.GetId() == orderId);
    }

    public void AddOrder(Order order)
    {
        orders.Add(order);
    }
}